src-tauri/src/api/geocoding.rs

```diff
@@ -13,18 +13,28 @@ fn construct_api_url(location: &str) -> String {
     )
 }
 
-pub async fn get_coordinates(location: &str) -> Result<GeocodingResult> {
-    let api_url = construct_api_url(location);
+pub async fn get_coordinates(location: &str) -> Result<GeocodingResult, errors::GeoCodingError> {
+    let api_url = construct_api_url(location.trim());
 
-    let response = reqwest::get(api_url)
-        .await?
-        .json::<GeocodingResponse>()
-        .await?
-        .results
-        .pop();
+    let response = match reqwest::get(api_url).await {
```

Comment:
Why return a custom error here? If you do `await?` it will automatically return the error anyway, I don't see a point here in defining our own errors. This also goes for the two match statements under this one.

Reply:
Well, one thing is it aids in handling errors in a structured way as can be seen in the calling function. So to aid in the overall expressiveness of our code. `await?` does return the error but we'll have to handle it with the error message in the calling function. This makes it more uniform.

Reply:
Fair enough, if you use `thiserror` as mentioned before, you can actually convert external errors into our own error types as described in [this StackOverflow post](https://stackoverflow.com/a/58337971). 
I think that'd be the best of both worlds as it saves us having to manually handle the errors with a `match` for every `Result` while having custom error that we can use. What do you think?

Reply:
I just updated with some changes. Please let me know what you think. I tried dabbling a bit into the thiserror crate and noticed there are some attributes e.g.:`#[from]`. Can these be handy in any way?

Reply:
I'm not sure, I haven't read too much into how we can rebind errors using `#[from]`. 
I think the code looks good, we can refine later how we want to handle the errors in more detail, since I'm not that familiar with how we should do custom errors in Rust myself.